tests/MavenTests/testMzAligner.cpp

```diff
@@ -44,15 +35,65 @@ void TestMzAligner::testDoAlignment() {
     QVERIFY(true);
 }
 
-void TestMzAligner::testSaveFit(){
+void TestMzAligner::testObiWarp()
+{
+
+    MavenParameters* mavenparameters = new MavenParameters;
+    mavenparameters->compoundMassCutoffWindow->setMassCutoffAndType(5,"ppm");
+    mavenparameters->minQuality = 0.8;
+    mavenparameters->minIntensity = 10000;
+    mavenparameters->matchRtFlag = false;
+    mavenparameters->samples = samples->alignmentSamples;
+    mavenparameters->clsf = new ClassifierNeuralNet;
+    mavenparameters->clsf->loadModel("bin/default.model");
+
+    ObiParams params("cor", false, 2.0, 1.0, 0.20, 3.40, 0.0, 20.0, false, 0.60);
+    Aligner aligner;
+    aligner.alignWithObiWarp(mavenparameters->samples, &params);
+
+    DBS.loadCompoundCSVFile("bin/methods/KNOWNS.csv");
+    vector<Compound*> compounds = DBS.getCopoundsSubset("KNOWNS");
+
+
+    PeakDetector peakDetector;
+    peakDetector.setMavenParameters(mavenparameters);
+    vector<mzSlice*> slices = peakDetector.processCompounds(compounds, "compounds");
+    peakDetector.processSlices(slices, "compounds");
 
-    vector<mzSample*> samplesToLoad;
-    for (int i = 0; i <  files.size(); ++i) {
-        mzSample* sample = new mzSample();
-        sample->loadSample(files.at(i).toLatin1().data());
-        samplesToLoad.push_back(sample);
+    for(PeakGroup grp: mavenparameters->allgroups) {
+
+        if(grp.getName() == "NAD+" || grp.getName() == "methionine" || grp.getName() == "dTMP") {
```

Comment:
The test condition looks good but I have concerns around the number of cases that will be covered using these 3 metabolites. Ideally, I would expect the following cases to be tested for overfitting and underfitting:
- Low and high intensity peaks in a group
- All high or low intensity peaks in a group
- Low resolution peaks

If we have such groups in these samples, then it's a small change. If not, let's think about changing the samples later.

Reply:
For unknown reasons, while running  MavenTests  the number of groups found by `processSlices` is way too less compared to the the groups found in GUI. If its not a too much of a problem, can we do this in the next release ?

Reply:
Could you check if this is the case in older releases as well? We can write tests for alignment after the release but if something in peak detection has changed in the core from the previous release, we should fix it. We have clients using CLI regularly.

Reply:
This has nothing to do with the CLI. My guess is that I am not setting the parameters correctly but I am not sure

Reply:
Cool. Leave this for now, then